### PR TITLE
test: Add test for free tx lane

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Adds a new test for a single free txn committed to a block via the free lane.

Uses the chaintestutil package (with the CreateTx PR https://github.com/skip-mev/chaintestutil/pull/4) and the network test suite.

Closes BLO-586

### CODE REVIEW

File: tests/integration/network_test.go [hunk 1]
```diff
@@ -116,3 +125,54 @@ func (s *NetworkTestSuite) QueryBlockSDKLanes() (*blocksdktypes.QueryLanesRespon
...
+	})
+	require.NoError(s.T(), err)
+	originalBalance := resp.Balance.Amount
+
+	// Send a free tx (delegation)
+	coin := sdk.NewCoin(s.NetworkSuite.Network.Config.BondDenom, math.NewInt(10))
+	txBz, err := s.NetworkSuite.CreateTxBytes(
+		context.Background(),
+		network.TxGenInfo{
+			Account:       *s.Accounts[0],
```
Comment (aljo242) [inline tests/integration/network_test.go:153]:
```suggestion
			Account:       acc,
```

File: tests/integration/network_test.go [hunk 2]
```diff
@@ -116,3 +125,54 @@ func (s *NetworkTestSuite) QueryBlockSDKLanes() (*blocksdktypes.QueryLanesRespon
...
+			GasLimit:      999999999,
+			TimeoutHeight: 999999999,
+			Fee:           sdk.Coins{coin},
```
Comment (aljo242) [inline tests/integration/network_test.go:156]:
```suggestion
			Fee:           sdk.NewCoins(coin),
```

**Review by aljo242 — APPROVED** [review]
tiny nits

File: tests/integration/network_test.go [hunk 3]
```diff
@@ -116,3 +125,54 @@ func (s *NetworkTestSuite) QueryBlockSDKLanes() (*blocksdktypes.QueryLanesRespon
 	client := blocksdktypes.NewQueryClient(cc)
 	return client.Lanes(context.Background(), &blocksdktypes.QueryLanesRequest{})
 }
+
+func (s *NetworkTestSuite) TestFreeTxNoFees() {
+	s.T().Parallel()
```
Comment (nivasan1) [inline tests/integration/network_test.go:130]: Avoid using `s.T().Parallel()` to avoid race-conditions in baseApp

**Review by nivasan1 — APPROVED** [review]
lgtm, is this rebased on main?

**Comment by Eric-Warehime** [comment]
> lgtm, is this rebased on main?

Updated--diff looks nice and compact now that all of our util stuff has landed. Going to push once CI succeeds.

**Pull request merged**